src/target/efm32.c

```diff
@@ -620,57 +583,55 @@ bool efm32_probe(target *t)
 	}
 
 	/* Read the part family, and reject if unknown */
-	size_t device_index  = efm32_lookup_device_index(t, di_version);
-	if (device_index > (127-32)) {
+	size_t device_index = efm32_lookup_device_index(t, di_version);
+	if (device_index > (127 - 32)) {
```

Comment:
the ()'s aren't required and both constants should be unsigned - perhaps manually simplify this with a comment on how such a magic value has been arrived at?

Reply:
I could not understand how this was arrived at, my best guess is it came from some sort of char conversion or so, either way, it didn't guard for an out of bound index for that table, so I changed it for its size

Reply:
ah! it comes from here

`	snprintf(priv_storage->efm32_variant_string, sizeof(priv_storage->efm32_variant_string), "%c\b%c\b%s %hu F%hu %s",
		di_version + 48, (uint8_t)device_index + 32, device->name, part_number, flash_kib, device->description);`

`(uint8_t)device_index + 32`

it is indeed some int to char conversion, but it looks like it was just checking if it was a valid char and not if it was out of bounds, this is used later in the cmd's to get the device with `efm32_get_device(t->driver[2] - 32);`

Reply:
the more I look at it the less sense it makes, I get that it's hiding the index in the driver string, and 32 (' ') is the first char that can be safely printed out the terminal, but surely there was a better way, even back when this was written